serverlessworkflow/modules/ROOT/pages/cloud/enhanced-service-discovery.adoc

```diff
@@ -0,0 +1,129 @@
+= Kubernetes Service Discovery for Workflow Application Configuration
+:compat-mode!:
+// Metadata:
+:description: Explain what is and how the service discovery works
+:keywords: kogito, workflow, quarkus, serverless, service-discovery, enhanced-service-discovery
+// links
+:quarkus_issue_url: https://github.com/quarkusio/quarkus/issues/27457
+
+
+The Kubernetes Service Discovery allows you to have a static URI that will define the Kubernetes resource that you use
+to perform http requests, the following sketch demonstrate how the SWUrl is composed:
+
+[source,shell]
+----
+kubernetes:<group>/<version>/<kind>/<namespace>/<name>?<attributeName>=<attributeValue>
+\________/ \_____/ \_______/ \____/ \_________/ \____/ \______________________________/
+  scheme    group   version   kind   namespace   name   additional resource attributes
+            \____________________/                       \__________________________/
+                Kubernetes GVK                                  Supported value: port_name
+----
+
+The following scheme values are supported:
+
+* **kubernetes**
+* **openshift**
+* **knative**
+
+For the GVK, the following Kubernetes Resources are currently supported:
+
+* **v1/service**
+* **serving.knative.dev/v1/service**
+* **v1/pod**
+* **apps/v1/deployment**
+* **apps.openshift.io/v1/deploymentconfig**
+* **apps/v1/statefulset**
+* **route.openshift.io/v1/route**
+* **networking.k8s.io/v1/ingress**
+
+[IMPORTANT]
+====
+The **GVK** (Group, Version and Kind) and the **Name** values are **mandatory** and can't be empty, otherwise, the engine will throw a validation issue and the discovery will not be performed.  +
```

Comment:
I think we can have "special" cases where the engine should assume the resource type. For Kubernetes/OpenShift, `deployment`, and for Knative, `ksvc`. 

Can you open a follow-up JIRA?

Reply:
That's might be a good idea :)
https://issues.redhat.com/browse/KOGITO-7863